instrumentation/action_mailer/lib/opentelemetry/instrumentation/action_mailer/instrumentation.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module ActionMailer
+      # The Instrumentation class contains logic to detect and install the ActionMailer instrumentation
+      class Instrumentation < OpenTelemetry::Instrumentation::Base
+        MINIMUM_VERSION = Gem::Version.new('6.1.0')
+        install do |_config|
+          resolve_email_address
+          ecs_mail_convention
+          require_dependencies
+        end
+
+        present do
+          defined?(::ActionMailer)
+        end
+
+        compatible do
+          gem_version >= MINIMUM_VERSION
+        end
+
+        option :disallowed_notification_payload_keys, default: [], validate: :array
+        option :notification_payload_transform,       default: nil, validate: :callable
+        option :email_address,                        default: :omit, validate: %I[omit include]
+
+        private
+
+        def gem_version
+          ::ActionMailer.version
+        end
+
+        def resolve_email_address
+          return unless ActionMailer::Instrumentation.instance.config[:email_address] == :omit
+
+          ActionMailer::Instrumentation.instance.config[:disallowed_notification_payload_keys] += ['email.to.address', 'email.from.address', 'email.cc.address', 'email.bcc.address']
+        end
+
+        def ecs_mail_convention
```

Comment:
Can we set this at install instead of checking with watch email?

Reply:
🤦🏼 it is already

Reply:
Do you mean calling this function inside `install do |_config| ... end` block?